web/src/narrow_error.ts

```diff
@@ -1,6 +1,12 @@
 import render_empty_feed_notice from "../templates/empty_feed_notice.hbs";
 
-export function narrow_error(narrow_banner_data) {
+interface NarrowErrorData {
+    title: string;
+    html: string;
+    search_data: string;
```

Comment:
`html` should be optional; `search_data` [is not a string](https://github.com/zulip/zulip/blob/main/web/src/narrow_banner.js#L137) and it is also optional.

Reply:
I've seen that search data has a function type which returns void. I'll be working on it asap. Thanks for your review

Reply:
I think `search_data` is not a function. It should be whatever `search_string_result`'s type is:
https://github.com/zulip/zulip/blob/b21d93cdd2d3647835b1b1201ba5e41872357605/web/src/narrow_banner.js#L25-L65.

Reply:
Im looking at the retrieve search function and it returns an array of objects, a boolean or a string, Im working on it

Reply:
wait I messed up my commits